lib/parser/livesplit_core_parser.rb

```diff
@@ -25,25 +25,42 @@ def self.parse(run, fast: false)
       program = ExchangeFormat.to_s
     end
 
+    key_error_lambda = ->(_hash, key) { raise KeyError, "#{key} not found" }
+
     run = parse_result.unwrap
+    run_metadata = run.metadata
 
     run_object = {
       program: program,
       game: run.game_name,
+      game_icon: run.game_icon.presence,
       category: run.category_name,
-      srdc_id: run.metadata.run_id,
+      name: run.extended_name(false),
+      metadata: {
+        srdc_id: run_metadata.run_id.presence,
+        platform_name: run_metadata.platform_name.presence,
+        region_name: run_metadata.region_name.presence,
+        uses_emulator: run_metadata.uses_emulator,
+        variables: {}
+      },
       attempts: run.attempt_count,
-      offset: run.offset.total_seconds,
-      history: nil,
-      indexed_history: nil,
-      realtime_history: nil,
+      offset_ms: run.offset.total_seconds * 1000,
+      history: [],
       splits: [],
-      realtime_time: 0,
-      gametime_time: 0,
+      realtime_duration_ms: 0,
+      gametime_duration_ms: 0,
       realtime_sum_of_best_ms: 0,
       gametime_sum_of_best_ms: 0,
       total_playtime_ms: 0
-    }.tap { |run_tap| run_tap[:name] = "#{run_tap[:game]} #{run_tap[:category]}".strip }
+    }
+    run_object.default_proc = key_error_lambda
+    run_object[:metadata].default_proc = key_error_lambda
```

Comment:
Is there a better way of setting up the `KeyError` proc on all nested hashes? I'd be concerned about adding another nested hash later and forgetting to do this vital thing of setting the default proc.

Either way looks like `run_object[:metadata][:variables]` doesn't have the proc set.

Reply:
I had initially avoided adding one on `run_object[:metadata][:variables]` since there are no guarantees that there are going to be any keys present (relies on the user to set them, so any or none can be present), but I think that adding the key safety and relying on whoever is checking that hash to be careful with access to it is better after thinking about it some more.